scripts/Fig5A-B_nb-umap-panels.R

```diff
@@ -0,0 +1,120 @@
+# This script generates UMAP panels 5A and 5B
+
+# load project
+renv::load()
+
+library(SingleCellExperiment)
+library(ggplot2)
+
+# set default ggplot theme for UMAPs
+theme_set(
+  theme_classic() +
+    theme(
+      strip.background = element_rect(fill = "transparent", linewidth = 0.5),
+      # no axis ticks or labels
+      axis.ticks = element_blank(),
+      axis.text = element_blank(),
+      aspect.ratio = 1
+    )
+)
+
+# Set up -----------------------------------------------------------------------
+
+project_id <- "SCPCP000004"
+data_dir <- here::here("s3_files", project_id)
+
+merged_sce_file <- file.path(data_dir, glue::glue("{project_id}_merged.rds"))
+merged_sce_df <- readr::read_rds(merged_sce_file) |>
+  # convert to data frame immediately for space
+  scuttle::makePerCellDF(
+    use.coldata = c(
+      "cell_id",
+      "library_id",
+      "infercnv_total_cnv",
+      "openscpca_celltype_annotation"
+    ),
+    use.dimred = "UMAP"
+  ) |>
+  # remove rownames
+  tibble::as_tibble()
+
+# define output files
+output_panel_a_file <- here::here("figures", "pdfs", "Fig5A-umap-celltypes.pdf")
+output_panel_b_file <- here::here("figures", "pdfs", "Fig5B-umap-infercnv.pdf")
+
+
+# Prepare data for plotting ----------------------------------------------------
+# Collapse cell types
+merged_sce_df <- merged_sce_df |>
+  dplyr::mutate(
+    cell_category = dplyr::case_when(
+      openscpca_celltype_annotation == "Neuroendocrine" ~ "malignant",
+      openscpca_celltype_annotation %in%
+        c("Unknown", "openscpca-excluded") ~ "unknown",
+      openscpca_celltype_annotation %in%
+        c(
+          "Fibroblast",
+          "Endothelial",
+          "Schwann"
+        ) ~ openscpca_celltype_annotation,
+      # everything else is a type of immune cell
+      .default = "immune cell"
+    ),
+    # tack on "cell" in a few spots
+    cell_category = ifelse(
+      cell_category %in% c("Endothelial", "Schwann", "malignant"),
+      glue::glue("{cell_category} cell"),
+      cell_category
+    ),
+    # make everything lowercase, consistent with other figures
+    cell_category = stringr::str_to_lower(cell_category),
+    # determine factor order based on frequency, with Unknown last
+    cell_category = forcats::fct_infreq(cell_category),
+    cell_category = forcats::fct_relevel(cell_category, "unknown", after = Inf)
+  )
+
+# define nbatlas condensed palette
+cell_palette <- c(
+  "malignant cell" = "rosybrown4",
+  "immune cell" = "darkgreen",
+  "endothelial cell" = "#FC1CBF",
+  "fibroblast" = "#B10DA1",
+  "schwann cell" = "chocolate1",
+  "unknown" = "gray70" # darker than validation palette since we have some standalone unknown clumps
+)
```

Comment:
I know it's only being used here, but I find it helpful to have the palettes as tsvs in the same folder so that if we need to add additional figures and re-use or modify this palette it's easy to find and use.